### PR TITLE
[HOTFIX] Forward compatibility of Note class

### PR DESCRIPTION
**Pull request by jongyoul**
### What is this PR for?
Solving inconsistency issue of Note's dateformat between branch-0.7 and master. It's related to #2272 

### What type of PR is it?
[Hot Fix]

### Todos
* [x] - Support several dateformat for Note class

### What is the Jira issue?
* N/A

### How should this be tested?
* Sync with some remote repository

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jongyoul** [comment]
@Leemoonsoo This is hotfix. Please test it and leave comments.

**Comment by zjffdu** [comment]
@jongyoul I hit this issue before, I think just adding `yyyy-MM-dd HH:mm:ss.SSS` into `NotebookImportDeserializer` should be able to resolve this issue.

**Comment by jongyoul** [comment]
Yes, if we use that gson, it can be solved. But there're several gson instances and used different one. I fixed it in order to use the same one to handle that type.

**Comment by jongyoul** [comment]
Some code used only vanila gson for handling json.

File: zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/ParagraphTest.java [hunk 1]
```diff
@@ -232,4 +226,11 @@ public void returnUnchangedResultsWithDifferentUser() throws Throwable {
 
 
   }
+
+  @Test
+  public void deserializeParagraphFromDifferentDateFormat() {
+    Paragraph paragraph = Note.GSON
+        .fromJson("{\"dateUpdated\":\"2017-09-01 00:01:02.345\"}", Paragraph.class);
+    System.out.println(paragraph.dateUpdated);
```
Comment (anthonycorbacho) [inline zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/ParagraphTest.java]: Can we have a better unit test for that? i dont think `System.out.println` is enough.
can we use something like `assert`?

**Comment by zjffdu** [comment]
@jongyoul Which code path use vanila gson for handling json ?

**Comment by jongyoul** [comment]
@zjffdu I found S3,VFS,Zeppelinhub repo used different gson from Note. I know it's fixed in master but need to fix them in 0.7.x branch.

**Comment by zjffdu** [comment]
Thanks @jongyoul

**Comment by jongyoul** [comment]
Merge if there's no more discussion

**Pull request closed without merging**